sdkproject/Assets/Mapbox/Unity/MeshGeneration/LayerVisualizers/LocationPrefabsLayerVisualizer.cs

```diff
@@ -0,0 +1,309 @@
+
+			if (item.findByType == LocationPrefabFindBy.MapboxCategory)
+			{
+				List<LocationPrefabCategories> categoriesList = GetSelectedCategoriesList(item.categories);
+				if (categoriesList == null || categoriesList.Count==0)
+					return;
+
+				List <string>stringsList = new List<string>();
+				var concatenatedString = "";
+
+				foreach (var category in categoriesList)
+				{
+					stringsList = LocationPrefabCategoryOptions.GetMakiListFromCategory(category);
+					if(string.IsNullOrEmpty(concatenatedString))
+						concatenatedString = string.Join(",", stringsList.ToArray());
+					else
+						concatenatedString += "," + string.Join(",", stringsList.ToArray());
+				}
+
+				LayerFilter filter = new LayerFilter(LayerFilterOperationType.Contains)
+				{
+					Key = propertyName,
+					PropertyValue = concatenatedString
+				};
+				AddFilterToItem(item, filter);
+			}
+		}
+
+		/// <summary>
+		/// Sets the density filters.
+		/// </summary>
+		/// <param name="item">Item.</param>
+		private void SetDensityFilters(PrefabItemOptions item)
+		{
+			if (item.density >= maxDensity) // decided that the max value for density
+				return;
+			
+			string propertyName = "";
+			item.densityPropertyFromFindByTypeDictionary.TryGetValue(item.findByType, out propertyName);
+
+			if (item.findByType == LocationPrefabFindBy.MapboxCategory || item.findByType == LocationPrefabFindBy.POIName)
+			{
+				LayerFilter filter = new LayerFilter(LayerFilterOperationType.IsLess)
+				{
+					Key = propertyName,
+					Min = item.density
+				};
+				AddFilterToItem(item, filter);
+			}
+		}
+
+		/// <summary>
+		/// Sets the name filters.
+		/// </summary>
+		/// <param name="item">Item.</param>
+		private void SetNameFilters(PrefabItemOptions item)
+		{
+			if (string.IsNullOrEmpty(item.nameString))
+				return;
+			
+			string propertyName = "";
+			item.namePropertyFromFindByTypeDictionary.TryGetValue(item.findByType, out propertyName);
+
+			if (item.findByType == LocationPrefabFindBy.POIName)
+			{
+				LayerFilter filter = new LayerFilter(LayerFilterOperationType.Contains)
+				{
+					Key = propertyName,
+					PropertyValue = item.nameString
+				};
+				AddFilterToItem(item, filter);
+			}
+		}
+
+		/// <summary>
+		/// Merges the filters with item filters.
+		/// </summary>
+		/// <param name="item">Item.</param>
+		void AddFilterToItem(PrefabItemOptions item, LayerFilter filter)
+		{
+			if (item.filterOptions == null)
+			{
+				item.filterOptions = new VectorFilterOptions();
+			}
+
+			item.filterOptions.filters.Add(filter);
+			item.filterOptions.combinerType = item._combinerType;
+
+		}
+
+		/// <summary>
+		/// Gets the list of categories selected through the dropdown
+		/// </summary>
+		/// <returns>The selected categories list.</returns>
+		/// <param name="cat">Cat.</param>
+		private List<LocationPrefabCategories> GetSelectedCategoriesList(LocationPrefabCategories cat)
```

Comment:
Can we come up with a more descriptive (but unfortunately less cute) name than "cat"?